### PR TITLE
add samtools/index nf-test test

### PR DESCRIPTION
**Pull request by maxulysse**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/samtools/index/tests/main.nf.test.snap [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+    "CRAI": {
+        "content": null,
```
Comment (SPPearce) [inline modules/nf-core/samtools/index/tests/main.nf.test.snap:3]: Why is this content null? Is this actually testing anything?

File: modules/nf-core/samtools/index/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,86 @@
+nextflow_process {
+
+    name "Test Process SAMTOOLS_INDEX"
+    script "../main.nf"
+    process "SAMTOOLS_INDEX"
+    tag "modules"
+    tag "modules_nfcore"
+    tag "samtools/index"
```
Comment (sateeshperi) [inline modules/nf-core/samtools/index/tests/main.nf.test:8]:
```suggestion
    tag "samtools"
    tag "samtools/index"
```

File: modules/nf-core/samtools/index/tests/main.nf.test [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+    test("BAI") {
```
Comment (sateeshperi) [inline modules/nf-core/samtools/index/tests/main.nf.test:10]:
```suggestion
    test("sarscov2 [BAI]") {
```

File: modules/nf-core/samtools/index/tests/main.nf.test [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        then {
+            assertAll (
+                { assert process.success },
+                { assert snapshot(process.out.index).match() },
+                { assert path(process.out.versions.get(0)).getText().contains("samtools") }
+            )
+        }
+    }
+
+    test("CRAI") {
```
Comment (sateeshperi) [inline modules/nf-core/samtools/index/tests/main.nf.test:35]:
```suggestion
    test("homo_sapiens [CRAI]") {
```

File: modules/nf-core/samtools/index/tests/main.nf.test [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        then {
+            assertAll (
+                { assert process.success },
+                { assert snapshot(process.out.index).match() },
+                { assert path(process.out.versions.get(0)).getText().contains("samtools") }
+            )
+        }
+    }
+
+    test("CSI") {
```
Comment (sateeshperi) [inline modules/nf-core/samtools/index/tests/main.nf.test:60]:
```suggestion
    test("homo_sapiens [CSI]") {
```

**Review by sateeshperi — APPROVED** [review]

**Comment by sateeshperi** [comment]
just noticed Simon's comment that snapshots are empty

File: modules/nf-core/samtools/index/tests/main.nf.test [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+                    file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true)
+                ]
+                """
+            }
+        }
+
+        then {
+            assertAll (
+                { assert process.success },
+                { assert snapshot(process.out.index).match() },
```
Comment (sateeshperi) [inline modules/nf-core/samtools/index/tests/main.nf.test:30]:
```suggestion
                { assert snapshot(process.out.index).match("bai") },
```

File: modules/nf-core/samtools/index/tests/main.nf.test [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+                    file(params.test_data['homo_sapiens']['illumina']['test_paired_end_recalibrated_sorted_cram'], checkIfExists: true)
+                ]
+                """
+            }
+        }
+
+        then {
+            assertAll (
+                { assert process.success },
+                { assert snapshot(process.out.index).match() },
```
Comment (sateeshperi) [inline modules/nf-core/samtools/index/tests/main.nf.test:55]:
```suggestion
                { assert snapshot(process.out.index).match("crai") },
```

File: modules/nf-core/samtools/index/tests/main.nf.test [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
```
Comment (sateeshperi) [inline modules/nf-core/samtools/index/tests/main.nf.test:82]:
```suggestion
                { assert snapshot(process.out.index).match("csi") },
```

**Review by sateeshperi — CHANGES_REQUESTED** [review]
with multiple tests need identifiers in snapshot

**Review by sateeshperi — APPROVED** [review]

File: modules/nf-core/samtools/index/tests/main.nf.test.snap [hunk 9]
```diff
@@ -0,0 +1,41 @@
+{
+    "csi": {
+        "content": [
+            [
+                [
+                    {
+                        "id": "test"
+                    },
+                    "test.paired_end.sorted.bam.csi:md5,8d63373007553e74d823fc2b9cbcf84d"
```
Comment (maxulysse) [inline modules/nf-core/samtools/index/tests/main.nf.test.snap:9]:
```suggestion
                    "test.paired_end.sorted.bam.csi:md5,3dd9e3ed959fca075b88bb8dc3cf7dbd"
```

File: modules/nf-core/samtools/index/tests/main.nf.test.snap [hunk 10]
```diff
@@ -0,0 +1,41 @@
+{
+    "csi": {
+        "content": [
+            [
+                [
+                    {
+                        "id": "test"
+                    },
+                    "test.paired_end.sorted.bam.csi:md5,3dd9e3ed959fca075b88bb8dc3cf7dbd"
```
Comment (maxulysse) [inline modules/nf-core/samtools/index/tests/main.nf.test.snap:9]:
```suggestion
                    "test.paired_end.sorted.bam.csi:md5,8d63373007553e74d823fc2b9cbcf84d"
```

**Pull request merged**